quality_of_service_interactive_demo/publisher_with_qos/publisher.cpp

```diff
@@ -0,0 +1,365 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <atomic>
+#include <cctype>
+#include <condition_variable>
+#include <iostream>
+#include <iomanip>
+#include <memory>
+#include <mutex>
+#include <thread>
+
+#ifdef _WIN32
+#include <conio.h>
+#else
+#include <termios.h>
+#endif
+
+#include "std_msgs/msg/string.hpp"
+#include "rcutils/cmdline_parser.h"
+#include "rclcpp/rclcpp.hpp"
+
+using namespace std::chrono_literals;
+
+//
+// helper function
+//
+double rmw_time_to_seconds(const rmw_time_t & time)
```

Comment:
You should move these utilities into a shared implementation, `rmw_time_to_seconds`, `CommandGetter` - anything shared between the two. Move to a header and add docstrings